crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -590,6 +609,22 @@ impl Entities {
         self.meta.get_unchecked_mut(index as usize).location = location;
     }
 
+    /// Increments the generation of a despawned [`Entity`]. This may only be called on an entity
+    /// index corresponding to an entity that once existed and has since been freed.
```

Comment:
How about something like this?
```suggestion
    /// Increments the `generation` of a despawned `index`. Next spawned entities with this `index` will count `generation` starting from the specified value + 1.
    ///
    /// Does nothing if such an entity with this `index` has not yet been despawned yet.
```

Reply:
Especially given the changes I've just made to `EntityMapper` I don't know if I want to use spawn/despawn language in here. That seems to imply a bit more work happening than what actually happens now. I'll use alloc/free to match the other methods in `Entities`

Reply:
Also, `starting from the specified value + 1` isn't technically true because the generation may have already been nonzero when this method was called. I'll clarify that